### ISSUE TITLE
Ambiguous device name/identifier ios9 xcode 7

### ISSUE THREAD

**Issue by hardik-d** [appium/appium#5619]
get following error:

 Failure/Error: Unable to find matching line from backtrace
 Selenium::WebDriver::Error::WebDriverError:
   Instruments crashed on startup

appium server logs:
info: [debug] [INST STDERR] Instruments Usage Error : Ambiguous device name/identifier 'iPhone 6 (9.0)':
iPhone 6 (9.0) [705F77E8-654F-405A-B9BB-549B727A8609]
iPhone 6 (9.0) + Apple Watch - 38mm (2.0) [9B3EE868-D843-4495-BD1F-7BDBA4B7EAA8]
info: [debug] [INSTSERVER] Instruments exited with code 255
info: [debug] Killall instruments
info: [debug] Instruments crashed on startup
.
..
....

my caps:

{caps: {
platformName: "iOS",
platformVersion: "9.0",
deviceName: "iPhone 6",
autoDismissAlerts: "true",
app: "#{ENV["APP_PATH"]}"
}

NOTE: I am running appium 1.4.11

**Comment by fangmobile** [comment]
same here.
change to iphone 5s will work.

**Comment by Jonahss** [comment]
Are you using appium version 1.4.11?

cc @moizjv

**Comment by moizjv** [comment]
@scottdixon also had this issue
```
iPhone 6 (9.0) [705F77E8-654F-405A-B9BB-549B727A8609]
iPhone 6 (9.0) + Apple Watch - 38mm (2.0) [9B3EE868-D843-4495-BD1F-7BDBA4B7EAA8]
```
is the problem

**Comment by moizjv** [comment]
@hardik-d @fangmobile can you use udid instead `705F77E8-654F-405A-B9BB-549B727A8609`, we have fixed this issue in appium 1.5?

**Comment by Jonahss** [comment]
Ok, as a workaround you can specify the `udid` in the desired capabilities.

this should be fixed in appium v1.5 so you can install the beta if you want and help us find errors :)
https://discuss.appium.io/t/appium-1-5-beta-release/6058

**Comment by hardik-d** [comment]
I will get beta

**Comment by fangmobile** [comment]
@moizjv I also noticed apple watch was the problem. that is why I switch to 5s and things are fine.
Another way is delete the simulator I guess.
I never used --udid before and getting ideviceinstaller problem.
Shouldn't ideviceinstaller be built already when I install appium js ?

**Comment by hardik-d** [comment]
issues installing beta, posted errors https://discuss.appium.io/t/appium-1-5-beta-release/6058. moving to udid approach for now

**Comment by hardik-d** [comment]
ran into the same ideviceinsatller issue and trying to install it messed my entire system. @Jonahss any possibility of a point release since this is basic appium startup failure?

**Comment by Jonahss** [comment]
OH, I see what's going on. Using `udid` makes appium do real-device stuff which will fail without ideviceinstaller. We will package ideviceinstaller into appium but it's a bit tricky.

What happens when you do `deviceName: '705F77E8-654F-405A-B9BB-549B727A8609'`

**Comment by Jonahss** [comment]
up to @jlipps if we want to change the logic we use to get simulator device names, or wait for appium 1.5 beta to be fully released. @moizjv said it would take a bit of work.

**Comment by Jonahss** [comment]
Yeah I think we should prioritize this.

**Comment by jlipps** [comment]
sounds like maybe it'd be a good idea to publish a quick fix to the device name logic so we ignore apple watch by default?

**Comment by hardik-d** [comment]
@jlipps i don't know what fix is implemented in 1.5 beta but what you mentioned would work for me. Please let me know when this fix would be in master so i can plan accordingly. Thanks!

**Comment by pavelzag** [comment]
Subbed for updates, experiencing the same issue

**Comment by iamchenchen** [comment]
experiencing the same thing, but quickest workaround for me is change deviceName to iPhone 6s
try this:
```python
    desired_caps = {
        'deviceName': 'iPhone 6s',
        'platformName': 'iOS',
        'platformVersion': '9.0',
        'app': PATH('../../apps/' + app),
    }
```

**Comment by mortocks** [comment]
Same issue

**Comment by Jonahss** [comment]
Yup, the fix is in and we're running tests for another patch release now.

**Comment by hardik-d** [comment]
thanks guys!

**Comment by amirwaisman** [comment]
Any news on this? When should the fix be out (and will it be available for appium.app? (GUI))

**Comment by maxmohash** [comment]
@jlipps the same issue is seen with ios simulator 9.1 we dont wish to hack the ios.js to add the next version all the time there is upgrade. can we expect a permanent solution to it without altering ios.js for simulator configurations on our own manually. Plus as we run the tests on saucelabs we shouldn't be fiddling with node_modules to somehow affect anything.

**Comment by imurchie** [comment]
What version of Appium are you using to run 9.1? This ought to be fixed.

**Comment by maxmohash** [comment]
Yeah it is handled thanks. One more thing will instruments-without-delay be fixed for XCode 7.1 or we have to use native instruments lib from now on.

**Comment by imurchie** [comment]
There is a fix. Details are here: https://github.com/appium/appium/blob/master/docs/en/advanced-concepts/iwd_xcode7.md

**Comment by maxmohash** [comment]
@imurchie many thanks

**Comment by jessejoe** [comment]
I am still seeing this issue with Appium `1.4.13` and Xcode 7:
```
$ appium --version
1.4.13
```

```python
In [22]: driver = webdriver.Remote('http://localhost:4723/wd/hub', {'deviceName': 'iPhone 6s', 'app': os.environ.get('MYAPP'), 'platformName': 'iOS'})
...
WebDriverException: Message: A new session could not be created. (Original error: Instruments crashed on startup)
```

```
info: Launching instruments
info: [debug] Attempting to run app on iPhone 6s (9.1)
info: On xcode 7.0, instruments-without-delay does not work, skippinginstruments-without-delay
info: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /tmp/appium-instruments/instrumentscli0.trace -w "iPhone 6s (9.1)" /Users/jesse/Library/Developer/Xcode/DerivedData/MyApp-gahhesvuzvbwttaqenfxtppsnygy/Build/Products/Debug-iphonesimulator/MyApp.app -e UIASCRIPT "/Users/jesse/Library/Application Support/appium/bootstrap/bootstrap-0fd5f45a68127c3d.js" -e UIARESULTSPATH /tmp/appium-instruments
info: [debug] And launch timeouts (in ms): {"global":90000}
info: [debug] [INST STDERR] Instruments Usage Error : Ambiguous device name/identifier 'iPhone 6s (9.1)':
iPhone 6s (9.1) [7985D8E6-719D-4284-90EA-EF2DA32242A4]
iPhone 6s (9.1) + Apple Watch - 42mm (2.0) [27B04B7E-84DC-416E-8FC0-87ED735A6A28]
```

**Comment by hardik-d** [comment]
what if you specified platformVersion: "9.1" in your caps

**Comment by jessejoe** [comment]
Same result with:
```python
In [18]: driver = webdriver.Remote('http://localhost:4723/wd/hub', {'deviceName': 'iPhone 6s', 'app': os.environ.get('MYAPP'), 'platformName': 'iOS', 'platformVersion':'9.1'})
```

**Comment by imurchie** [comment]
Can you provide the whole Appium server log (preferably as a link to a [gist](https://gist.github.com/))?

**Comment by hardik-d** [comment]
i have upgraded to 1.4.15 now and works there as well! hope you are not doing this from appium.io or still starting the old appium version

**Comment by jessejoe** [comment]
I was using `1.4.13`, but I tried upgrading to `1.4.16` and still see the same issue. Here's the full server log - https://gist.github.com/jessejoe/55936433854515316619

**Comment by imurchie** [comment]
Ah! Yes, the logic at the moment only translates `iPhone 6` and `iPhone Simulator`. Not sure why. @moiz?

Two quick-and-dirty solutions: (1) get rid of the device with the watch attached; (2) use a `deviceName` of `=iPhone 6s (9.1) [` (with the `=` at the beginning (this will skip the device name translation, and give you a device that Instruments can boot).

**Comment by hardik-d** [comment]
Even i didn't realize i am running test on iphone 6 not 6s, you could just do that as well

**Comment by jessejoe** [comment]
```
(2) use a deviceName of =iPhone 6s (9.1) [ (with the = at the beginning (this will skip the device name translation, and give you a device that Instruments can boot).
```

This is a nice workaround and works, thanks! But is this bug still valid?

**Comment by imurchie** [comment]
The bug is fixed for `iPhone 6`. Other devices still will fail.

**Comment by ThermoKnight** [comment]
Seeing same behavior as @imurchie.  iPhone 6 works but iPhone 6s fails.

**Comment by toejough** [comment]
if this is still a bug for devices in general, can we re-open this issue?

**Comment by SrinivasanTarget** [comment]
@imurchie @scottdixon Can we close this? This was fixed in 1.5.
